core/build.gradle

```diff
@@ -9,3 +9,8 @@ dependencies {
     testCompile 'com.google.code.gson:gson:2.5' // for example
     testCompile 'org.springframework:spring-context:4.2.5.RELEASE' // for example
 }
+
+configure(compileTestJava) {
+    sourceCompatibility = 1.8
```

Comment:
there's an existing issue on source level, and looks like our last call was to re-evaluate when JDK 9 was out...
https://github.com/Netflix/feign/issues/289